packages/lodestar/src/sync/regular/naive/naive.ts

```diff
@@ -154,12 +169,43 @@ export class NaiveRegularSync implements IRegularSync {
   }
 
   private getSyncPeers = async (): Promise<PeerId[]> => {
-    return this.network.getPeers().reduce( (validPeers: PeerId[], peer: PeerId) => {
-      const rep = this.reps.getFromPeerId(peer);
-      if(rep && rep.latestStatus) {
-        validPeers.push(peer);
+    if (!this.network.getPeers().includes(this.bestPeer)) {
+      // bestPeer disconnected
+      this.bestPeer = undefined;
+      await this.waitForBestPeer();
+    }
+    return [this.bestPeer];
+  };
+
+  private waitForBestPeer = async (): Promise<void> => {
```

Comment:
Can we make this cleaner like have two functions:
getBestHead(currentSlot, peers, reps): {root, slot} // highest slot
getBestPeer(root, reps): PeerId // takes all peers with this headRoot and get one with best score (once we land new gossipsub, take random for now)



Those methods should be standalone so we can test them properly